src/main/scala/inox/solvers/SolverFactory.scala

```diff
@@ -109,6 +110,7 @@ object SolverFactory {
           replacement
 
         case Some(_) => name
+        case None if name.startsWith("smt-z3:") => name
```

Comment:
Should we maybe check that the solver binary is available using
```scala
val hasZ3Exec = try {
  new Z3Interpreter(name.drop(7), Array("-in", "-smt2"))
  true
} catch {
  case _: java.io.IOException => false
}
```
and throw a fatalError if it's unavailable?

Also, maybe extract the `name.drop(7)` part into `getZ3Executable(name)`?

Reply:
Sure, I've added this